### PR TITLE
Try running verification on GHA

### PR DESCRIPTION
**Pull request by nealrichardson**

### CODE REVIEW

**Comment by kszucs** [comment]
I'm not sure yet how we're going to integrate this into the release workflow, but having the builds passing for this branch with RC2 hardcoded would certainly help with the release vote!

File: .github/workflows/rc.yml [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  push:
+    paths:
+      - '.github/workflows/rc.yml'
+      # This is just a file we know will get edited when a version is bumped
+      # and shouldn't get touched otherwise
+      - 'ruby/red-arrow/lib/arrow/version.rb'
+  pull_request:
+    paths:
+      - '.github/workflows/rc.yml'
+      - 'ruby/red-arrow/lib/arrow/version.rb'
```
Comment (pitrou) [inline .github/workflows/rc.yml]: Shouldn't this be a cron job instead? Or is the goal to run this _only_ when bumping the version?

**Comment by pitrou** [comment]
I'm actually not sure if this is useful. Ideally we would have a way to continuously run source release verification _before_ producing and uploading tarballs manually?

File: .github/workflows/rc.yml (hunk 1 above)
Comment (nealrichardson) [reply .github/workflows/rc.yml]: Probably so; this was my first time running these scripts so I wasn't sure how much of them depended on installing rc artifacts from a published repository, in which case we couldn't do them nightly unless we also published nightly artifacts (not a bad idea but out of scope).

**Comment by nealrichardson** [comment]
Closing in favor of #6836

**Pull request closed without merging**